core/src/main/java/org/n52/iceland/ogc/gml/time/TimePosition.java

```diff
@@ -62,8 +54,7 @@ public TimePosition(DateTime time) {
      *            Indeterminate value of time position
      */
     public TimePosition(TimeIndeterminateValue indeterminateValue) {
-        super();
-        this.indeterminateValue = indeterminateValue;
+        this.indeterminateValue = Optional.of(indeterminateValue);
```

Comment:
The setters use the same call ``Optional.of`` - and imho it is OK to get an exception when doing ``setTimeFormat(null)``.